dipy/workflows/vis_registeration.py

```diff
@@ -0,0 +1,278 @@
+                slice_mosaic.display(None, None, cnt)
+                slice_mosaic.SetPosition((X + border) * i,
+                                         0.5 * cols * (Y + border) -
+                                         (Y + border) * j, 0)
+                slice_mosaic.SetInterpolate(False)
+                renderer.add(slice_mosaic)
+                renderer.reset_camera()
+                renderer.zoom(1.6)
+                cnt += 1
+                if cnt > num_slices:
+                    break
+            if cnt > num_slices:
+                break
+
+        renderer.reset_camera()
+        renderer.zoom(1.6)
+
+        window.record(renderer, out_path=fname,
+                      size=(900, 600), reset_camera=False)
+
+    def adjust_color_range(self, img):
+
+        """
+        Function to adjust the range of colors in numpy array
+        to create the GIF (GIF standard only supports 256 colours).
+
+        return
+        The numpy array with scaled down range of color values.
+        """
+
+        # Interpolating to range 0-15 for reducing the colours.
+        img[..., 0] = np.interp(img[..., 0], (img[..., 0].min(),
+                                              img[..., 0].max()), (0, 14))
+        img[..., 1] = np.interp(img[..., 1], (img[..., 1].min(),
+                                              img[..., 1].max()), (0, 14))
+        img = np.round(img, 0).astype('uint8')
+
+        # Interpolating back to the range 0-255 for getting GIF range.
+        img[..., 0] = np.interp(img[..., 0], (img[..., 0].min(),
+                                              img[..., 0].max()), (0, 255))
+        img[..., 1] = np.interp(img[..., 1], (img[..., 1].min(),
+                                              img[..., 1].max()), (0, 255))
+
+        img = np.round(img, 0).astype('uint8')
+
+        return img
+
+    def animate_overlap_with_renderer(self, static_img, moved_img,
+                                      sli_type, fname, affine):
+
+        """
+        Function for creating the animated GIF from the slices of the
+        registered image.
+        It uses the renderer object to control the dimensions of the
+        created GIF and for correcting the orientation.
+        Patameters
+        ----------
+        sli_type : str (optional)
+            the type of slice to be extracted:
+            sagital, coronal, axial, None (default).
+
+        fname: str, optional
+            Filename for saving the GIF (default 'animation.gif').
+        """
+
+        overlay, value_range = self.process_image_data(static_img,
+                                                       moved_img)
+        x, y, z, _ = overlay.shape
+
+        num_slices = 0
+
+        if sli_type == 'saggital':
+            num_slices = x
+            slice_type = 0
+        elif sli_type == 'coronal':
+            num_slices = y
+            slice_type = 1
+        elif sli_type == 'axial':
+            num_slices = z
+            slice_type = 2
```

Comment:
It's `sagittal`. Also, what happens when it's something else than those three names? Ok, you write `None (default).` but it looks like this function is doing nothing on `None`.

Reply:
Yup, it is 'sagittal', corrected the typo. Also, earlier I was checking the 'None' inside the functions and returning immediately if that was true. Later I decided to check for this in the run method but forgot to remove the 'None' from the function documentation. Good point, though. Thanks.